ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/StreamedQueryResourceTest.java

```diff
@@ -138,14 +142,40 @@ public void setup() {
     testResource = new StreamedQueryResource(
         ksqlConfig,
         mockKsqlEngine,
-        serviceContext,
+        () -> serviceContext,
+        mockStatementParser,
+        commandQueue,
+        DISCONNECT_CHECK_INTERVAL,
+        COMMAND_QUEUE_CATCHUP_TIMOEUT,
+        activenessRegistrar);
+  }
+
+  private StreamedQueryResource givenStreamedQueryResource(final ServiceContext serviceContext) {
```

Comment:
same comment as above, this should take in a `Supplier<ServiceContext>` and we should call it from above

Reply:
done